### PR TITLE
Suspend "opposing" watcher during build process

### PR DESCRIPTION
**Pull request by mlange-42**
When saving multiple times shortly after another, it may happen that the attempted reverse operation fails due to "Source/code files changed. Stopping to avoid overwriting..."

Thus, simply checking for changes before writing is not sufficien. Suspending the "opposing" watcher also completely avoids the unneccessary attempt for the opposing build direction.

### CODE REVIEW

File: src/watch.rs [hunk 1]
```diff
@@ -73,6 +86,20 @@ fn update_watcher(
     Ok(())
 }
 
+fn unwatch_all(watcher: &mut RecommendedWatcher, old_files: &HashSet<PathBuf>) -> Fallible {
+    for path in old_files {
+        watcher.unwatch(&path)?;
```
Comment (adamreichold) [inline src/watch.rs:91]: This is probably a rather costly operation as it involves multiple system calls. A more lightweight approach would probably be to share an `suspend: Arc<AtomicBool>` with each background thread which leads to all events being ignored when `suspend.load(Ordering::SeqCst)` is true.

Comment (mlange-42) [reply src/watch.rs:91]: I will give it a try, but not sure if this works with the *debounce* watcher that delays by 1 second.

Comment (mlange-42) [reply src/watch.rs:91]: With an implementation as in #165, it seems not to work. Maybe I missed something, but I think it is due to the delay in debounce watcher.

Comment (adamreichold) [reply src/watch.rs:91]: (As an aside, do you really need the debounced watcher as you already coalesce events anyway? You might want to wait for quiescence though, i.e. reset the timeout whenever an event arrives as there is little use to rebuilding while the files still change.)

**Comment by mlange-42** [comment]
Closed in favor of #165

**Pull request closed without merging**